### PR TITLE
Implements GNIP for curated-thumbs app

### PR DESCRIPTION
**Pull request by capooti**
This PR implements the GNIP for curated-thumbs #4726 

Here is an animated gif with the new feature:
http://g.recordit.co/JaKutevbrT.gif

It has an additional requirement: django-imagekit

Thumbnails are implemented for layers and maps. It would be very easy to implement them for documents as well but I see that there is already this feature (let me know if you want to replace it with this for homogeneity).

I decided to implement the application as an internal application for GeoNode as I believe this is useful for any GeoNode instance. If you believe it should rather go to a contrib application, I am ready to migrate it to geonode-contribs.

### CODE REVIEW

**Comment by afabiani** [comment]
... moreover, I don't see any test

File: geonode/curated_thumbs/tests.py [hunk 1]
```diff
@@ -0,0 +1 @@
+# -*- coding: utf-8 -*-
```
Comment (afabiani) [inline geonode/curated_thumbs/tests.py]: Missing headers

Comment (afabiani) [inline geonode/curated_thumbs/tests.py]: Missing tests

**Comment by capooti** [comment]
Thanks Alessio
having this in Base definitely makes sense. I will also include tests and documentation.
I am closing this PR and will send a PR
cheers

**Pull request closed without merging**